### PR TITLE
Update rebalancing

### PR DESCRIPTION
**Pull request by al1img**

### CODE REVIEW

File: cmserver/cmserver_test.go [hunk 1]
```diff
@@ -126,8 +126,12 @@ func TestConnection(t *testing.T) {
 	}
 
 	statusFotaNotification := cmserver.UpdateFOTAStatus{
-		Components:   []cloudprotocol.ComponentStatus{{ComponentID: "1234", Version: "123.4321"}},
-		UnitConfig:   &cloudprotocol.UnitConfigStatus{Version: "bc_version"},
+		Components: []cloudprotocol.ComponentStatus{
+			{
+				ComponentID:   "testComponent",
+				ComponentType: "testType", Version: "2.0.0",
```
Comment (mlohvynenko) [inline cmserver/cmserver_test.go:132]:
```suggestion
				ComponentType: "testType", 
				Version: "2.0.0",
```

File: unitconfig/unitconfig.go [hunk 2]
```diff
@@ -176,14 +168,24 @@ func (instance *Instance) load() (err error) {
...
+		return aoserrors.Wrap(err)
+	}
+
+	if newVer.Equal(curVer) {
+		return ErrAlreadyInstalled
 	}
 
-	return unitConfig.Version, nil
+	if newVer.LessThan(curVer) {
+		return aoserrors.New("wrong version")
```
Comment (mlohvynenko) [inline unitconfig/unitconfig.go:187]: Optional,
consider returning more specific error, for example "version downgrade is prohibited"

File: unitstatushandler/unitstatushandler.go [hunk 3]
```diff
@@ -461,77 +463,73 @@ func (descriptor *statusDescriptor) getVersion() (version string) {
...
 	instance.statusMutex.Lock()
 	defer instance.statusMutex.Unlock()
 
 	log.WithFields(log.Fields{
-		"id":            componentInfo.ComponentID,
-		"status":        componentInfo.Status,
-		"vendorVersion": componentInfo.Version,
-		"error":         componentInfo.ErrorInfo,
+		"id":            status.ComponentID,
+		"status":        status.Status,
```
Comment (mlohvynenko) [inline unitstatushandler/unitstatushandler.go:490]: Let's log component type as well
```suggestion
		"type":        status.ComponentType,
```

**Review by MykolaSolyanko — APPROVED** [review]

File: unitconfig/unitconfig.go (hunk 2 above)
Comment (al1img) [reply unitconfig/unitconfig.go:187]: Will keep as is.

File: unitstatushandler/unitstatushandler.go (hunk 3 above)
Comment (al1img) [reply unitstatushandler/unitstatushandler.go:490]: done

**Review by mykola-kobets-epam — APPROVED** [review]

**Review by mlohvynenko — APPROVED** [review]

**Pull request merged**